### PR TITLE
Add contour plot function

### LINKED ISSUE #12034: Add contour plot function
(Mentioned on [gitter](https://gitter.im/sympy/sympy?at=58740606cbcb2817704aa5dd), no responses yet.)

I saw 2d plots mentioned on [this wiki page](https://github.com/sympy/sympy/wiki/Plotting-capabilities#types-of-plots), but I understand that might be outdated/inaccurate.

Looking through what `plot3d()` does, I used `Plot(ContourSeries(…)).show()` as a workaround.
It seems like much of the infrastructure is in place to support a contour plotting function, would there be any difficulty doing so?

### PR DESCRIPTION
**Pull request by ArifAhmed1995**
Adds contour plotting capability.

Fixes #12034

### CODE REVIEW

**Comment by ArifAhmed1995** [comment]
@asmeurer Can you review this PR?

**Comment by asmeurer** [comment]
@chrstphrchvz can you take a look?

File: sympy/plotting/plot.py [hunk 1]
```diff
@@ -1730,6 +1730,26 @@ def plot3d_parametric_surface(*args, **kwargs):
...
 
+def conplot(*args, **kwargs):
+    """
+    Draws contour plot of a given function.
+    """
+    args = list(map(sympify, args))
+    symbols = set()
+    show = kwargs.pop('show', True)
+    plot_expr = check_arguments(args, 1, 2)
+    symbols |= plot_expr[0].free_symbols
```
Comment (asmeurer) [inline sympy/plotting/plot.py]: What is the point of this. You set `symbols` to an empty set and then you set it to `free_symbols`. Why not just check `free_symbols`? Did you intend to write `symbols = kwargs.pop('symbols', set())`?

Comment (asmeurer) [inline sympy/plotting/plot.py]: And at any rate, shouldn't it be required for the symbols to be specified, since they need to be in order?

Comment (ArifAhmed1995) [inline sympy/plotting/plot.py]: Yes , some part of the code wasn't required. I've modified it. 
All symbols that are part of the plot expression are independent variables so I don't see why they need to be in order.The code in `def plot3d` does not require it to be so. Or maybe I've misunderstood what you mean by order.

**Comment by ArifAhmed1995** [comment]
@chrstphrchvz Thanks for the review ! 
I'll implement the stuff you mentioned next week as my mid semester exams begin tomorrow.

**Comment by asmeurer** [comment]
`plot_contour` is a better name. That matches the other plotting functions in the module.

**Comment by asmeurer** [comment]
For the example, usually you'd want something with a lot of hills and valleys. Like `sin(x)*sin(y)`, or something similar.

**Comment by gxyd** [comment]
@ArifAhmed1995 can you please fix the merged conflicts, and address the requested changes?

**Comment by ArifAhmed1995** [comment]
@gxyd Sure ! I'm a little busy this week, but think I can take out some time for this PR.

**Pull request closed without merging**